app/controllers/PayFrequencyController.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.PayFrequencyFormProvider
+import javax.inject.Inject
+import models.NormalMode
+import navigation.Navigator
+import pages.PayFrequencyPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.PayFrequencyView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PayFrequencyController @Inject()(
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  navigator: Navigator,
+  getSession: GetSessionAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: PayFrequencyFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: PayFrequencyView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController with I18nSupport {
+
+  private val form = formProvider()
+
+  def onPageLoad(): Action[AnyContent] = (getSession andThen getData andThen requireData) { implicit request =>
```

Comment:
what is `requireData`? what's the difference between that and `getData`?

Reply:
`requireData` enforces that some UserAnswers should present at this stage in mongo, for the existing session, if not it will throw session expired

```class DataRequiredActionImpl @Inject()(implicit val executionContext: ExecutionContext) extends DataRequiredAction {

  override protected def refine[A](request: OptionalDataRequest[A]): Future[Either[Result, DataRequest[A]]] =
    request.userAnswers match {
      case None =>
        Future.successful(Left(Redirect(routes.SessionExpiredController.onPageLoad())))
      case Some(data) =>
        Future.successful(Right(DataRequest(request.request, request.identifier, data)))
    }
}```

Reply:
`getData`  transforms the request into OptionalDataRequest 
```
class DataRetrievalActionImpl @Inject()(val sessionRepository: SessionRepository)(implicit val executionContext: ExecutionContext)
    extends DataRetrievalAction {

  override protected def transform[A](request: IdentifierRequest[A]): Future[OptionalDataRequest[A]] =
    sessionRepository
      .get(request.identifier)
      .map(OptionalDataRequest(request.request, request.identifier, _))
}
```

Reply:
`getData` is `ActionTransformer[IdentifierRequest, OptionalDataRequest]` 
while `requireData` is `ActionRefiner[OptionalDataRequest, DataRequest]`

Reply:
except the first page all other pages should have this `requireData` restriction